### PR TITLE
Add print rust

### LINKED ISSUE #1993: Implement `Print` cheat code in rust
We should add the code for the `Print` cheat code.
The implementation should be put [here](https://github.com/software-mansion-labs/cairo/blob/a8ca26f2c50cab560c2e2ac98c6a8deabb332996/crates/cairo-lang-runner/src/casm_run/mod.rs#L774).

We were considering using [starknet_in_rust](https://github.com/lambdaclass/starknet_in_rust) and [blockifier](https://github.com/starkware-libs/blockifier), and we decided to go with the latter.

### PR DESCRIPTION
**Pull request by karol-bisztyga**
### Checklist
- [x] Relevant issue is linked
- [ ] Docs updated/issue for docs created
- [x] Added relevant tests

https://github.com/software-mansion-labs/cairo/pull/89

### CODE REVIEW

File: protostar-rust/run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  exit 0;
+fi
+
+for DIR in */ ; do
+  pushd "$DIR"
+  cargo run
+  popd # $DIR
+done
+
+popd # cairo-tests
```
Comment (cptartur) [inline protostar-rust/run_tests.sh:20]: We shouldn't have to use scripts like this. See e2e tests in #2033. This should be rebased.

**Review by cptartur — CHANGES_REQUESTED** [review]

File: protostar-rust/cairo-tests/declare_test/protostar.toml
Comment (piotmag769) [inline protostar-rust/cairo-tests/declare_test/protostar.toml:1]: We got rid of protostar.toml in the new runner

File: protostar-rust/cairo-tests/print_test/tests/simple_test.cairo
Comment (piotmag769) [inline protostar-rust/cairo-tests/print_test/tests/simple_test.cairo:1]: Can we add tests for printing felts that:
a) do not fit into 31 bytes
b) fit into 31 bytes but are not decodable to ascii

We didn't handle this cases in python runner, currently being fixed in https://github.com/software-mansion/protostar/pull/2070, I think we can do it in a similar way here

File: pyrightconfig.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "ignore": ["./starknet_in_rust/", "./blockifier/"]
```
Comment (piotmag769) [inline pyrightconfig.json:2]: Why do we ignore `starknet_in_rust` here?

**Review by piotmag769 — CHANGES_REQUESTED** [review]
I assume a lot will change in this PR when suggestion from https://github.com/software-mansion/protostar/pull/2059 are applied, requesting changes just to make sure to review it later on

**Review by piotmag769 — APPROVED** [review]

**Review by cptartur — APPROVED** [review]

**Pull request merged**